server/app/mutations/stacks/create.rb

```diff
@@ -47,6 +61,26 @@ def execute
         return
       end
 
+      if self.volumes
+        # TODO Separate into own method
+        attributes.delete(:volumes)
+        self.volumes.each do |volume|
+          if volume[:external]
+            vol = Volume.where(name: volume['name'], grid: grid, stack: nil).first
+            if vol
```

Comment:
I find the `if` a bit odd here... if we've validated it, then can't we just assume it's not nil? It feels scary to have code that just leaves out a volume.

I guess I might setup some kind of of `@external_volumes[name] = vol` in the `validate_volumes` and then re-use those models here...